numpy/core/src/multiarray/nditer_pywrap.c

```diff
@@ -1418,12 +1438,6 @@ static PyObject *npyiter_value_get(NewNpyArrayIterObject *self)
         ret = npyiter_seq_item(self, 0);
     }
     else {
-        if (self->is_closed) {
-            PyErr_SetString(PyExc_ValueError,
-                    "Iterator is closed");
-            return NULL;
-        }
-
```

Comment:
What is the reason for removing all these `is_closed` checks? It is related to #10956, right?

Reply:
There is a check for `self->iter` above each of these code blocks. It will now be `NULL`, so the checks and the flag have been removed